src/main/java/teammates/logic/core/InstructorsLogic.java

```diff
@@ -174,6 +174,18 @@ public void verifyIsEmailOfInstructorOfCourse(String instructorEmail, String cou
         }
     }
 
+    public void verifyAtLeastOneInstructorIsDisplayed(boolean isOriginalInstructorDisplayed, String courseId,
+                                                      boolean isEditedInstructorDisplayed)
+            throws InvalidParametersException {
+        List<InstructorAttributes> instructorsDisplayed = instructorsDb.getInstructorsDisplayedToStudents(courseId);
+        boolean isEditedInstructorChangedToNonVisible = isOriginalInstructorDisplayed && !isEditedInstructorDisplayed;
+
+        if ((instructorsDisplayed.isEmpty())
```

Comment:
This is still a logic gap. `instructorsDisplayed.isEmpty() && !isOriginalInstructorDisplayed && isEditedInstructorDisplayed` should the update valid.